core/templates/dev/head/pages/questions_list/QuestionsListDirective.js

```diff
@@ -33,25 +33,34 @@ oppia.directive('questionsList', [
       templateUrl: UrlInterpolationService.getDirectiveTemplateUrl(
         '/pages/questions_list/questions_list_directive.html'),
       controller: [
-        '$scope', '$http', '$q', '$uibModal', '$window', 'AlertsService',
-        'QuestionCreationService', 'UrlService', 'NUM_QUESTIONS_PER_PAGE',
-        'EditableQuestionBackendApiService', 'EditableSkillBackendApiService',
-        'MisconceptionObjectFactory', 'QuestionObjectFactory',
-        'EVENT_QUESTION_SUMMARIES_INITIALIZED', 'StateEditorService',
-        'QuestionUndoRedoService', 'UndoRedoService', function(
-            $scope, $http, $q, $uibModal, $window, AlertsService,
-            QuestionCreationService, UrlService, NUM_QUESTIONS_PER_PAGE,
-            EditableQuestionBackendApiService, EditableSkillBackendApiService,
-            MisconceptionObjectFactory, QuestionObjectFactory,
-            EVENT_QUESTION_SUMMARIES_INITIALIZED, StateEditorService,
-            QuestionUndoRedoService, UndoRedoService) {
+        '$scope', '$filter', '$http', '$q', '$uibModal', '$window',
+        'AlertsService', 'QuestionCreationService', 'UrlService',
+        'NUM_QUESTIONS_PER_PAGE', 'EditableQuestionBackendApiService',
+        'EditableSkillBackendApiService', 'MisconceptionObjectFactory',
+        'QuestionObjectFactory', 'EVENT_QUESTION_SUMMARIES_INITIALIZED',
+        'StateEditorService', 'QuestionUndoRedoService', 'UndoRedoService',
+        function(
+            $scope, $filter, $http, $q, $uibModal, $window,
+            AlertsService, QuestionCreationService, UrlService,
+            NUM_QUESTIONS_PER_PAGE, EditableQuestionBackendApiService,
+            EditableSkillBackendApiService, MisconceptionObjectFactory,
+            QuestionObjectFactory, EVENT_QUESTION_SUMMARIES_INITIALIZED,
+            StateEditorService, QuestionUndoRedoService, UndoRedoService) {
           $scope.currentPage = 0;
 
           var _initTab = function() {
             $scope.questionEditorIsShown = false;
             $scope.question = null;
             $scope.questionSummaries =
               $scope.getQuestionSummaries($scope.currentPage);
+            if ($scope.questionSummaries) {
```

Comment:
This is dodgy. You shouldn't be adding (or modifying?) fields to a domain object midway.

Couldn't you either generate the truncated version on the fly, or pass the truncated version from the backend?

Reply:
Done, created a truncated version in the frontend.